### PR TITLE
[type: refactor]mark as deprecated.

### PR DESCRIPTION
**Pull request by KevinClair**
Make sure that:

- [ ] You have read the [contribution guidelines](https://shenyu.apache.org/community/contributor-guide).
- [ ] You submit test cases (unit or integration tests) that back your changes.
- [ ] Your local test passed `./mvnw clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/consul/ConsulDataInit.java [hunk 1]
```diff
@@ -25,7 +25,10 @@
 
 /**
  * The type Consul data init.
+ *
+ * @deprecated replace with {@link ConsulDataChangedInit}
  */
+@Deprecated
```
Comment (yu199195) [inline shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/consul/ConsulDataInit.java:31]: why add Deprecated

Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/consul/ConsulDataInit.java:31]: Replaced with `ConsulDataChangedInit`, this class will no longer be used.

Comment (yu199195) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/consul/ConsulDataInit.java:31]: this used by  consul sycn data？

Comment (KevinClair) [reply shenyu-admin/src/main/java/org/apache/shenyu/admin/listener/consul/ConsulDataInit.java:31]: yes.

**Review by yu199195 — APPROVED** [review]

**Pull request merged**